src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseEnvironmentMembers.cs

```diff
@@ -0,0 +1,134 @@
+                    !compilationContext.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemDiagnosticsProcess, out var processType) ||
+                    !compilationContext.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemDiagnosticsProcessModule, out var processModuleType) ||
+                    !compilationContext.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingThread, out var threadType))
+                {
+                    return;
+                }
+
+                // Get the various members needed from Environment, Process, ProcessModule, and Thread.
+                var environmentProcessIdSymbol = environmentType.GetMembers("ProcessId").FirstOrDefault();
+                var environmentProcessPathSymbol = environmentType.GetMembers("ProcessPath").FirstOrDefault();
+                var environmentCurrentManagedThreadIdSymbol = environmentType.GetMembers("CurrentManagedThreadId").FirstOrDefault();
+                var processGetCurrentProcessSymbol = processType.GetMembers("GetCurrentProcess").FirstOrDefault();
+                var processIdSymbol = processType.GetMembers("Id").FirstOrDefault();
+                var processMainModuleSymbol = processType.GetMembers("MainModule").FirstOrDefault();
+                var processModuleFileNameSymbol = processModuleType.GetMembers("FileName").FirstOrDefault();
+                var threadCurrentThreadSymbol = threadType.GetMembers("CurrentThread").FirstOrDefault();
+                var threadManagedThreadId = threadType.GetMembers("ManagedThreadId").FirstOrDefault();
+
+                // This one analyzer looks for multiple things, so figure out which we have the symbols to enable.
+                bool lookForProcessId = processGetCurrentProcessSymbol != null && processIdSymbol != null && environmentProcessIdSymbol != null;
+                bool lookForProcessPath = processGetCurrentProcessSymbol != null && processMainModuleSymbol != null && processModuleFileNameSymbol != null && environmentProcessPathSymbol != null;
+                bool lookForCurrentManagedThreadId = threadCurrentThreadSymbol != null && threadManagedThreadId != null && environmentCurrentManagedThreadIdSymbol != null;
+                if (!lookForProcessId && !lookForProcessPath && !lookForCurrentManagedThreadId)
+                {
+                    return;
+                }
+
+                // Everything we're looking for is a property, so find all property references.
+                compilationContext.RegisterOperationAction(operationContext =>
+                {
+                    var initialPropRef = (IPropertyReferenceOperation)operationContext.Operation;
+
+                    // Process.GetCurrentProcess().Id
+                    if (lookForProcessId && processIdSymbol!.Equals(initialPropRef.Property))
```

Comment:
I think this should use SymbolEqualityComparer.Default

Reply:
> I think this should use SymbolEqualityComparer.Default

Why is that better than calling Equals?

Reply:
@stephentoub  I don't have full context to be honest. But that's what RS1024 analyzer recommends. The analyzer is currently disabled due to a crash bug, and will be re-enabled after it's fixed. If you enabled it, I believe you'll get a warning here.

Probably @sharwell or @mavasani have more context on why this is recommended.

Reply:
Happy to change it if it's the right thing to do. But would like to understand. @sharwell ?

Reply:
`SymbolEqualityComparer` allows the author to be explicit on whether `string` and `string?` should be considered the same type (`SymbolEqualityComparer.Default`) or different (`SymbolEqualityComparer.IncludeNullability`). Code written before Nullable Reference Types used `Equals` but frequently failed to account for the difference here. The analyzer is a disambiguation aid for future maintainers who would otherwise be unsure whether the analyzer was written with NRT in mind.

Reply:
So I should use it to avoid a future analyzer yelling at me, but there's no functional difference in this case nor is there an ambiguity since none of the symbols in question are impacted by NRT. Correct?

Reply:
Yes, that is the action item here

Reply:
as @sharwell says, using the default equals is fine is nullability isn't a part of symbol identity in your case. The analyzer warns here because we want analyzer authors to be aware of this and make it explicit which behavior they indend.